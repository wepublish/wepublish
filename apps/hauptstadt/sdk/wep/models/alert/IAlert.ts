export default interface IAlert {
  title: string
  type: 'error' | 'success' | 'primary'
}
