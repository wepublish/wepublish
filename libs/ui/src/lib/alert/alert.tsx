import {Alert as MuiAlert} from '@mui/material'
import {ComponentProps, PropsWithChildren} from 'react'

export type AlertProps = PropsWithChildren<ComponentProps<typeof MuiAlert>>

export function Alert({children, ...props}: AlertProps) {
  return <MuiAlert {...props}>{children}</MuiAlert>
}
