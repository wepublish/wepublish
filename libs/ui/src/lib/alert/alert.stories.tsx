import {Meta} from '@storybook/react'
import {Alert, AlertProps} from './alert'

export default {
  component: Alert,
  title: 'UI/Alert'
} as Meta<typeof Alert>

const Template = (args: AlertProps) => <Alert {...args}>Alert</Alert>

export const Default = {
  render: Template
}
