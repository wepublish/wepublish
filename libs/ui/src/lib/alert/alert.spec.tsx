import {composeStories} from '@storybook/react'
import {render} from '@testing-library/react'

import * as stories from './alert.stories'

const storiesCmp = composeStories(stories)

describe('Alert', () => {
  Object.entries(storiesCmp).forEach(([story, Component]) => {
    it(`should render ${story}`, () => {
      const {asFragment} = render(<Component />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
