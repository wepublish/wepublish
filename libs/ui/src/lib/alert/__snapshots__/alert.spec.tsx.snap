// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render Default 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-wnkddu-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="SuccessOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      Alert
    </div>
  </div>
</DocumentFragment>
`;
